src/cryptoadvance/specter/templates/includes/qr-scanner.html

```diff
@@ -104,16 +104,29 @@
         // expected: crypto-psbt
         let re3 = /UR:CRYPTO-(.*)/;
         let p3 = result.match(re3);
-        if(p3){
-          this.bcur2decoder.receivePart(result);
-          if(this.bcur2decoder.isComplete()){
-            result = this.bcur2decoder.resultUR();
-            this.progress.innerHTML = "";
-          }else{
-            this.progress.innerHTML = `Progress: ${Math.round(this.bcur2decoder.getProgress()*100)}%`;
-            return;
+
+        // check if it is a bcur2 Foundation Passport stuff
+        let re4 = /UR:BYTES\//;
+        let p4 = result.match(re4);
+
+        if (p3 || p4){
```

Comment:
If you add one more check:
```js
if (p3 || (p4 && !p2) ){
```
you will not need to handle multi-piece bcur1 strings, so in the exception you could treat your result as final right away.